alchemiscale/tests/integration/storage/test_statestore.py

```diff
@@ -1245,3 +1272,469 @@ def test_remove_scope_duplicate(
         scopes = n4js.list_scopes(user.identifier, credential_type)
         assert scope not in scopes
         assert not_removed in scopes
+
+    # status-setting function related tests.
+    # would be too much boilerplate to test all the functions individually
+    # so we parameterize over the methods
+
+    @pytest.fixture()
+    def f_set_task_running(self, n4js):
+        return n4js.set_task_running
+
+    @pytest.fixture()
+    def f_set_task_complete(self, n4js):
+        return n4js.set_task_complete
+
+    @pytest.fixture()
+    def f_set_task_error(self, n4js):
+        return n4js.set_task_error
+
+    @pytest.fixture()
+    def f_set_task_waiting(self, n4js):
+        return n4js.set_task_waiting
+
+    @pytest.fixture()
+    def f_set_task_invalid(self, n4js):
+        return n4js.set_task_invalid
+
+    @pytest.fixture()
+    def f_set_task_deleted(self, n4js):
+        return n4js.set_task_deleted
+
+    # allowed = [running, error, invalid, deleted]
+    # no-op = [waiting]
+    # not allowed = [complete]
+    @pytest.mark.parametrize(
+        "status_func, status",
+        [
+            ("f_set_task_running", TaskStatusEnum.running),
```

Comment:
Add in `failure_flag` to avoid specific pytest raises cases and  repeat testing of failures,